### PR TITLE
Migrate to Java 21

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,12 +1,12 @@
 org.gradle.caching=true
 group=io.ballerina.stdlib
 version=1.6.2-SNAPSHOT
-ballerinaLangVersion=2201.8.0
+ballerinaLangVersion=2201.10.0-SNAPSHOT
```
Comment (HindujaB) [inline gradle.properties:4]: Shall we generate a new lang timestamp with java21 and update it here? It will fix the build failure

Comment (HindujaB) [reply gradle.properties:4]: triggered a lang build https://github.com/ballerina-platform/ballerina-lang/actions/runs/11047985959

Comment (warunalakshitha) [reply gradle.properties:4]: Should we done that finally. PR Build will anyway use lang version which was build

Comment (HindujaB) [reply gradle.properties:4]: I think the PR build pick from the `gradle.properties`. So it might fail for `2201.10.0-SNAPSHOT`

Comment (warunalakshitha) [reply gradle.properties]: Updated with timestamp.

**Pull request closed without merging**